src/main/java/org/everit/osgi/ecm/component/webconsole/ECMWebConsoleServlet.java

```diff
@@ -113,6 +129,36 @@ public ECMWebConsoleServlet(
 
   }
 
+  private int[] addState(final int[] intArray, final ComponentState state) {
+    switch (state) {
+      case ACTIVE:
+        intArray[ACTIVE_POSITION]++;
```

Comment:
Please do not be such a C++ developer :-). Either use separate int member variables for the states or create a separate class that has member variables for all of the states. E.g.: public class ComponentStateSum { int activeSateSum, int failedStateSum, ...}. The HTML template code will be easier to read as well.